### PR TITLE
Added support for the "Split" op and added a VJP for the casting initializer.

### PR DESCRIPTION
**Pull request by eaplatanios**

### CODE REVIEW

**Comment by rxwei** [comment]
Could we make sure `Split` is tested or has a use case?

File: stdlib/public/TensorFlow/Tensor.swift [hunk 1]
```diff
@@ -206,11 +206,23 @@ public extension Tensor {
...
     self = Raw.cast(other)
   }
 }
 
+internal extension Tensor where Scalar : TensorFlowFloatingPoint {
+  @inlinable
+  static func _vjpCast<OtherScalar : TensorFlowFloatingPoint>(
+    _ other: Tensor<OtherScalar>
+  ) -> (Tensor<Scalar>, (Tensor<Scalar>) -> Tensor<OtherScalar>) {
+    return (Tensor<Scalar>(other), { v in Tensor<OtherScalar>(v) })
```
Comment (rxwei) [inline stdlib/public/TensorFlow/Tensor.swift]:
```suggestion
    return (Tensor(other), { v in Tensor<OtherScalar>(v) })
```

File: stdlib/public/TensorFlow/Tensor.swift [hunk 2]
```diff
@@ -206,11 +206,23 @@ public extension Tensor {
...
   init<OtherScalar : Numeric>(_ other: Tensor<OtherScalar>) {
     self = Raw.cast(other)
   }
 }
 
+internal extension Tensor where Scalar : TensorFlowFloatingPoint {
+  @inlinable
+  static func _vjpCast<OtherScalar : TensorFlowFloatingPoint>(
+    _ other: Tensor<OtherScalar>
+  ) -> (Tensor<Scalar>, (Tensor<Scalar>) -> Tensor<OtherScalar>) {
```
Comment (rxwei) [inline stdlib/public/TensorFlow/Tensor.swift]:
```suggestion
  ) -> (Tensor, (Tensor) -> Tensor<OtherScalar>) {
```

**Review by rxwei — COMMENTED** [review]
The cast part LGTM.

**Comment by eaplatanios** [comment]
> Could we make sure `Split` is tested or has a use case?

It's kind of dual to `SplitV` in that, if you want to split in equal parts (e.g., when "unstacking" a tensor along the batch dimension), you'd need to compute the size of each split and copy over an array with the sizes, if using `SplitV`. `Split` lets you do that by just providing the number of splits you want. I'm planning to add a `split` function to `Tensor` in the open PR in swift-apis soon. I have implemented them but need to add VJPs and then will push changes there.

**Comment by rxwei** [comment]
OK. I just want to make sure it's at least tested on your machine so that you won't need to come back and change it if it doesn't work.

**Comment by eaplatanios** [comment]
Yes that makes sense. Give me a couple hours maybe and I'll post here once I've tested it. :)

**Comment by eaplatanios** [comment]
@rxwei Just tested locally and it works fine. :)

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow

**Comment by rxwei** [comment]
@swift-ci please test tensorflow Linux

**Pull request merged**